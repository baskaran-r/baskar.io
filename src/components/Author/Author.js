// @flow strict
import { Link, withPrefix } from 'gatsby';
import React from 'react';
import { useSiteMetadata } from '../../hooks';
import { getContactHref } from '../../utils';

import styles from './Author.module.scss';

export const Author = ({ showBio, showTwitter, small }) => {
  const photoSizeWithOutBio = small ? 40 : 48;
  const photoSize = showBio ? 60 : photoSizeWithOutBio;
  const { author } = useSiteMetadata();

  return (
    <div className={styles['author']}>
      <div className={styles['author__header']}>
        <Link to="/" className={styles['author__header-photo']}>
          <img
            src={withPrefix(author.photo)}
            width={photoSize}
            height={photoSize}
            alt={author.name}
          />
        </Link>
        <div className={styles['author__header-right']}>
          <h4 className={styles['author__header-right-name']}>
            <Link
              className={styles['author__header-right-name-link']}
              to="/"
              rel="author"
            >
              {author.name}
            </Link>
            {showTwitter && ' '}
            {showTwitter && (
              <a
                style={{ fontWeight: 500 }}
                href={getContactHref('twitter', author.contacts.twitter)}
                target="_blank"
                rel="noopener noreferrer"
              >
                @{author.contacts.twitter}
              </a>
            )}
          </h4>
          {showBio && (
            <p
              className={styles['author__header-right-bio']}
              dangerouslySetInnerHTML={{ __html: author.bio }}
            />
          )}
        </div>
      </div>
    </div>
  );
};

export default Author;
