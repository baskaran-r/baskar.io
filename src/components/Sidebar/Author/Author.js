// @flow strict
import React from 'react';
import { withPrefix, Link } from 'gatsby';
import styles from './Author.module.scss';

type Props = {
  author: {
    name: string,
    bio: string,
    photo: string
  }
};

const Author = ({ author } : Props) => (
  <div className={styles['author']}>
    <div className={styles['author__main-section']}>
      <Link to="/">
        <img src={withPrefix(author.photo)} className={styles['author__photo']} alt={author.name} />
      </Link>
      <div className={styles['author__title']}>
        <h1>
          <Link className={styles['author__title-link']} to="/">
            {author.name}
          </Link>
        </h1>
      </div>
    </div>
    <p className={styles['author__subtitle']} dangerouslySetInnerHTML={{ __html: author.bio }} />
  </div>
);

export default Author;
