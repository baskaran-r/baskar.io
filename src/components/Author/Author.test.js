import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';

import { Author } from './Author';

describe('Author', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    const props = {
      author: {
        name: 'test',
        bio: 'test',
        photo: 'test',
      },
    };

    const tree = renderer.create(<Author {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
