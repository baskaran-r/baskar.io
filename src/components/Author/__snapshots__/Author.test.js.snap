// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div
  className="author"
>
  <div
    className="author__header"
  >
    <a
      className="author__header-photo"
      href="/"
    >
      <img
        alt="Test name"
        height={48}
        src="/test.jpg"
        width={48}
      />
    </a>
    <div
      className="author__header-right"
    >
      <h4
        className="author__header-right-name"
      >
        <a
          className="author__header-right-name-link"
          href="/"
          rel="author"
        >
          Test name
        </a>
      </h4>
    </div>
  </div>
</div>
`;
